Blish HUD/GameServices/GameIntegration/AudioIntegration.cs

```diff
@@ -1,38 +1,66 @@
+        private const string    APPLICATION_SETTINGS = "OverlayConfiguration";
+        private const string    USEGAMEVOLUME_SETTINGS = "GameVolume";
+        private const string    VOLUME_SETTINGS = "Volume";
+        private const string    DEVICE_SETTINGS = "OutputDevice";
+        private const int       CHECK_INTERVAL = 250;
+        private const int       AUDIO_DEVICE_UPDATE_INTERVAL = 5000;
+        private const int       AUDIOBUFFER_LENGTH = 20;
+        private const float     MAX_VOLUME = 0.4f;
+        private readonly        SettingEntry<bool> _useGameVolume;
+        private readonly        SettingEntry<Devices> _deviceSetting;
+        private readonly        MMDeviceEnumerator _deviceEnumerator;
+        private readonly        RingBuffer<float> _audioPeakBuffer = new RingBuffer<float>(AUDIOBUFFER_LENGTH);
+        private readonly        SettingEntry<float> _volumeSetting;
+        
+        private readonly List<(MMDevice AudioDevice, AudioMeterInformation MeterInformation)> _gw2AudioDevices = new List<(MMDevice AudioDevice, AudioMeterInformation MeterInformation)>();
 
         private double _timeSinceCheck = 0;
+        private double _timeSinceAudioDeviceUpdate = 0;
 
-        private float? _averageGameVolume;
+        private float? _volume;
 
         /// <summary>
-        /// Provides an estimated volume level for the application
-        /// based on the volumes levels exhibited by the game.
+        /// This either provides  an estimated volume level for the application
+        /// based on the volumes levels exhibited by the game
+        /// or
+        /// the set volume in settings.
         /// </summary>
-        public float AverageGameVolume => _averageGameVolume ??= CalculateAverageVolume();
+        public float Volume => _volume ??= GetVolume();
+
+        /// <summary>
+        /// Current used AudioDevice. This either the same as GW2 is using
+        /// or the selected one in the settings.
+        /// </summary>
+        public MMDevice AudioDevice { get; private set; }
 
         public AudioIntegration(GameIntegrationService service) : base(service) {
+            var audioSettings = GameService.Settings.RegisterRootSettingCollection(APPLICATION_SETTINGS);
+            _useGameVolume = audioSettings.DefineSetting(USEGAMEVOLUME_SETTINGS, true, Strings.GameServices.OverlayService.Setting_UseGameVolume_DisplayName, Strings.GameServices.OverlayService.Setting_UseGameVolume_Description);
+            _volumeSetting = audioSettings.DefineSetting(VOLUME_SETTINGS, MAX_VOLUME / 2, Strings.GameServices.OverlayService.Setting_Volume_DisplayName, Strings.GameServices.OverlayService.Setting_Volume_Description);
+            _volumeSetting.SetRange(0.0f, MAX_VOLUME);
+            _deviceSetting = audioSettings.DefineSetting(DEVICE_SETTINGS, Devices.Gw2OutputDevice, Strings.GameServices.OverlayService.Setting_AudioDevice_DisplayName, Strings.GameServices.OverlayService.Setting_AudioDevice_Description);
```

Comment:
Ownership for some for these changes may not be appropriate here.  It's hard to say.  We don't have a dedicated audio service, but this particular ServiceModule was specifically for providing the integration to pull from the game's audio levels with its estimates which is why it's owned by the `GameIntegrationService`.

It may be that we need to have an audio service or provide one as a ServiceModule under a rewritten `ContentService`.  🤔

In the meantime, it's probably fine to keep here until that work is done.

Reply:
#336 for the primary continuation of this.

Reply:
So the `AudioIntegration` purpose was mainly to get the ingame audio average volume and for nothing else? Would make sense bc before the "AudioService" from MonoGame handled the playing of the sound. Would also move out the whole audio playing in `ContentService` into that AudioService and move the whole thing i did out of the `AudioIntegration` into that service too tbh